Implement glue objects

haarchri opened the pull request:
Signed-off-by: Christopher Haar <chhaar30@googlemail.com>

### Description of your changes

added following glue services:

- classifier
- connection
- crawler
- database
- job
- securityconfiguration


#Fixes #522
I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

created & deleted all objects from example/glue & example/iam
overview:

```
kubectl get managed | grep "glue"
database.glue.aws.crossplane.io/glue-database   True    True     glue-database
classifier.glue.aws.crossplane.io/glue-classifier-csv    True    True     glue-classifier-csv
classifier.glue.aws.crossplane.io/glue-classifier-grok   True    True     glue-classifier-grok
classifier.glue.aws.crossplane.io/glue-classifier-json   True    True     glue-classifier-json
classifier.glue.aws.crossplane.io/glue-classifier-xml    True    True     glue-classifier-xml
securityconfiguration.glue.aws.crossplane.io/glue-securityconfiguration   True    True     glue-securityconfiguration
job.glue.aws.crossplane.io/glue-job   True    True     glue-job
crawler.glue.aws.crossplane.io/glue-crawler   True    True     glue-crawler
connection.glue.aws.crossplane.io/glue-connection   True    True     glue-connection
iamrole.identity.aws.crossplane.io/glue-role      True    True     8h
iamrolepolicyattachment.identity.aws.crossplane.io/glue-rolepolicyattachment                 True    True     glue-role      arn:aws:iam::255932642927:policy/glue-policy                 8h
iampolicy.identity.aws.crossplane.io/glue-policy   arn:aws:iam::255932642927:policy/glue-policy   True    True     8h

```

[contribution process]: https://git.io/fj2m9

haarchri commented:
fixed gofmt

haarchri commented:
i will have look for check-diff after rebase master

AaronME commented:
@haarchri Please rebase on master and we can run this through testing.

Thank you!

haarchri commented:
i will rebase in the next days thanks

AaronME reviewed (approved):
This looks good. All resources created and deleted successfully. Thanks, @haarchri!

Please resolve the remaining conflicts and we can merge.

haarchri commented:
@AaronME i will update shortly also with ref and selectors - i will ping you

haarchri commented:
@AaronME ready - can you have a look ? ref & selector implemented, rebased and adjusted the examples - also added iam stuff needed for gets glue running ;)

thanks for your time

AaronME commented on examples/glue/classifier.yaml:
```diff
@@ -0,0 +1,55 @@
+      rowTag: "timestamp"
+      classification: "xml"
+  providerConfigRef:
+    name: default
```
```suggestion
    name: example
```

AaronME commented on examples/glue/classifier.yaml:
```diff
@@ -0,0 +1,55 @@
+      disableValueTrimming: true 
+      allowSingleColumn: false
+  providerConfigRef:
+    name: default
```
```suggestion
    name: example
```

AaronME commented on examples/glue/classifier.yaml:
```diff
@@ -0,0 +1,55 @@
+    jsonClassifier: 
+      jsonPath: "[<timestampe>]"
+  providerConfigRef:
+    name: default
```
```suggestion
    name: example
```

AaronME commented on examples/glue/classifier.yaml:
```diff
@@ -0,0 +1,55 @@
+      grokPattern: "%{MY_TIMESTAMP} "
+      customPatterns: "MY_TIMESTAMP (%{USERNAME:test}[,]%{YEAR:year}[-]%{MONTHNUM:mm}[-]%{MONTHDAY:dd} %{TIME:time})"
+  providerConfigRef:
+    name: default
```
```suggestion
    name: example
```

AaronME commented on examples/glue/connection.yaml:
```diff
@@ -0,0 +1,22 @@
+        securityGroupIDRefs:
+          - name: sample-cluster-sg
+  providerConfigRef:
+    name: default
```
```suggestion
    name: example
```

AaronME commented on examples/glue/crawler.yaml:
```diff
@@ -0,0 +1,28 @@
+        connectionName: "glue-connection",
+      ]
+  providerConfigRef:
+    name: default
```
```suggestion
    name: example
```

AaronME commented on examples/glue/database.yaml:
```diff
@@ -0,0 +1,10 @@
+  forProvider:
+    region: us-east-1
+  providerConfigRef:
+    name: default
```
```suggestion
    name: example
```

AaronME commented on examples/glue/job.yaml:
```diff
@@ -0,0 +1,16 @@
+      pythonVersion: "3"
+      scriptLocation: "s3://aws-glue-crossplane-us-east-1/scripts"
+  providerConfigRef:
+    name: default
```
```suggestion
    name: example
```

AaronME commented on examples/glue/securityconfiguration.yaml:
```diff
@@ -0,0 +1,19 @@
+        kmsKeyARNRef:
+          name: dev-key
+  providerConfigRef:
+    name: default
```
```suggestion
    name: example
```

AaronME reviewed (commented):
@haarchri Works great!

Please update the ProviderConfigRefs to `example,` then I can approve and merge.

Also, I've noticed you are not implementing `isUpToDate`, which means your resources can only be created and destroyed. Is this intentional?

Thank you for all your contributions!

AaronME commented:
@haarchri do not worry about rebasing when you make the next changes.

haarchri commented:
@AaronME i will have a look in a followup for `isUpToDate` - not all glue services can be updated ..

AaronME commented:
@haarchri I attempted to resolve a conflict with a previous merge and it appears to be causing some test failures.

@muvaf - let's sync on this to figure out what this PR needs. Passed testing before my commit.

haarchri commented:
@AaronME  can you have a look now ? i changed the ratelimiter & run regeneration 
commit: https://github.com/crossplane/provider-aws/pull/686/commits/c1e250b6164d68019f0f75525ab5f0e749db4f00

on local machine make reviewable and make check-diff is working 
did you want to have a look ? 
thanks

haarchri commented:
@AaronME all checks passed ;)
